### PR TITLE
fix: trash.cmd defaults to 'trash' on macos

### LINKED ISSUE #1366: Warn users when trash command is not available
### Description

If `gio trash` is not available on the user's system, attempting to trash a file instead permanently deletes the file with no warning to the user. For anyone who was using `trash` before, after #1323 they would now be deleting files without knowing.

### Neovim version

```text
NVIM v0.8.0-dev+1894-gc57f6b28d
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

macOS 13.0 Beta

### nvim-tree version

79258f1

### Minimal config

```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      "kyazdani42/nvim-tree.lua",
      "kyazdani42/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  require("nvim-tree").setup {}
end
```

### Steps to reproduce

1. touch delete-me.txt
2. nvim -nu nvt-min.lua
3. `:NvimTreeToggle`
4. hover "delete-me.txt" and press `D` (confirming the deletion)

### Expected behavior

Pressing `D` to trash a file should either trash the file, in this case `gio trash` must be available, or fail the operation.

### Actual behavior

Pressing `D` to trash a file will permanently a file if `gio trash` is not available.

### PR DESCRIPTION
**Pull request by alex-courtis**
fixes #1366

### CODE REVIEW

File: lua/nvim-tree.lua [hunk 1]
```diff
@@ -663,6 +663,13 @@ local function validate_options(conf)
   end
 end
 
+--- Apply OS specific localisations to DEFAULT_OPTS
+local function localise_default_opts()
+  if utils.is_macos then
+    DEFAULT_OPTS.trash.cmd = "trash"
```
Comment (gegoune) [inline lua/nvim-tree.lua:669]: Shouldn't that be `gio trash`?

Comment (alex-courtis) [reply lua/nvim-tree.lua:669]: This is where I get confused. I was under the impression that gio would put the files in `~/.local/share/Trash/files` which is not recognised by macos.

Perhaps we should just update the doc, removing "Trash is unavailable" and noting the available homebrew options for macos users.

Comment (alex-courtis) [reply lua/nvim-tree.lua:669]: I think it might be time for me to use https://github.com/sickcodes/Docker-OSX

Comment (kennyishihara) [reply lua/nvim-tree.lua:669]: [Here](https://github.com/nvim-tree/nvim-tree.lua/issues/1366#issuecomment-1646788804), we found that neither `trash` nor `gio` comes preinstalled on macOS so it's best to do what you said, which is to warn the users that `D` is not available unless you install `trash` from homebrew. We shouldn't use `gio trash` on macOS as it moves the file to a different location than `~/.Trash` which is where the Bin icon is pointing to.

**Comment by alex-courtis** [comment]
Updated to include the windows trash changes #2335

- [ ] @linrongbin16 what is the value of `trash.cmd` for windows?

**Comment by linrongbin16** [comment]
Hi, @alex-courtis, there's no such *default* trash command on Windows.
People have to install third party command to make it work. For now, the most porpular one is: https://github.com/sindresorhus/trash-cli. The command is just: `trash`.

**Comment by alex-courtis** [comment]
> 

Thanks mate. Adding to the doc.

**Comment by linrongbin16** [comment]
sure thing

**Review by linrongbin16 — APPROVED** [review]
lgtm

**Pull request merged**